WIP update to work with MXLIMS 0.6.9 (rhfogh: rhfogh_mxlims_2025)

rhfogh opened the pull request:
Cleaned up. refactored, newer JSON schema and tool versions. See Release notes for details

rhfogh committed:
WIP Working on MXExperiment generation

rhfogh committed:
Finished create_mxexperiment function

rhfogh committed:
Bug fixes and WIP improving MXLIMS handling

rhfogh committed:
Model bug fixes -> 0.2.4. Should now support  phasing experiments (?)

rhfogh committed:
Minor model change -> 0.2.5. Now support GPhL characterisation and interleaving experiments

rhfogh committed:
Added line indent to final MXLIMS JSON output

rhfogh committed:
Added updated mxlims record export

rhfogh committed:
Updated model,generation. amd use. WIP

rhfogh committed:
Documentation fix

rhfogh committed:
Fixed pydantic/html generation

rhfogh committed:
Debugged export of MXLIMS json data. Now works

rhfogh committed:
WIP trying to get MXCuBE impl and Pydantic to work

rhfogh committed:
Added and tested MXLIMS export

rhfogh committed:
Added mxlims loading test code

rhfogh committed:
Linting clean-up

rhfogh committed:
Made mxcubecore/configuration/mockup.yaml and adapted mockups to use it

rhfogh committed:
Updated mxcubeweb config for gphl

rhfogh committed:
Fixed yaml linting problems

rhfogh committed:
Fixed yaml linting problems 2

rhfogh committed:
Adapted to MXLIMS model changes

rhfogh committed:
Adapted to MXLIMS 0.6.4 and tested

rhfogh committed:
Ruff reformatting

rhfogh committed:
Satisfied Ruff

rhfogh committed:
Fixed bug in QtGraphicsMAnager.init

rhfogh committed:
Bug fix - removed debug printout

rhfogh committed:
Updated to use new MXLIMS output code

rhfogh committed:
removed now-obsolete imports

rhfogh committed:
Removed no-longer-used configuration directory

rhfogh committed:
Made MXLIMS export function again; Updated poetry.lock

rhfogh committed:
Fixed som ruff complaints

marcus-oscarsson commented:
Is this a standing PR to sort of visualize whats being done with MXLIMS in a MXCuBE context or do you intend to merge this relatively soon ? Basically whats the plan for this PR ;)

rhfogh commented:
@marcus-oscarsson I would like to merge this, but I think the right time would be when someone other than me wanted to use some of it, and so could help check if it works for them. As it stands it is more, like you say, 'a standing PR to sort of visualize whats being done with MXLIMS in a MXCuBE context' or indeed just how one would integrate MXLIMS. I would hope soon to get to the point where this is being integrated at some beamline, at which point I shall remove the WIP.

marcus-oscarsson commented:
Ok, sounds good

rhfogh committed:
linter nitpicking

rhfogh committed:
Bug fix in MXLIMS export

rhfogh committed:
Some fixes for GPhL - MXLIMS integration

rhfogh committed:
Updated to MXLIMS 0.6.10

rhfogh committed:
Fixed typo

rhfogh committed:
Removed &SEGMENT_LIST from instrumentation.nml

rhfogh committed:
Commented out exotic strategies

rhfogh committed:
Fixes for kappa-limits, use-for-indexing

rhfogh committed:
Bug fix in latest commit

rhfogh committed:
Changed "GΦL" to "GPhL" in all strings

rhfogh committed:
Fix and clean-up after introduction of new AbstractDiffractometer.

This includes converting in_kappa_mode and in_plate_mode to properties everywhere, and removing user_confirm_centring and its mentions, that were not actually used anywhere.

rhfogh committed:
Changed move_omega_relative to omega.set_value_relative, and improved GPhL log output

rhfogh committed:
Cleaned up (get_)head_type

rhfogh committed:
Fiddling to satisfy ruff

rhfogh committed:
Fixed ruff nits, and globally disactivated ICN001 ruff rule.

This rule, making it illegal to do e.g. 'import numpy' rather than 'import numpy as np' is broken in lots of places each of which had its own disabling statement. Personally I think it is pointless, but I will be happy to comply with it, provided the person who wants to activate it will go throlugh the code and fix all occurrences first.

rhfogh committed:
Addiotional linting - import order, typos and types

rhfogh committed:
More ruff linting for import order

rhfogh committed:
CHanged MXLIMS timestamp to UTC

rhfogh committed:
Import and code cleanup - renamed utils/mxlims to utils/mxutils
